pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -58,6 +72,14 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		}
 	}
 
+	// Note that an empty list won't be parsed as nil.
+	if obj.TopologySpreadConstraints == nil {
```

Comment:
distinguishing between nil and [] is not generally desired... is there a reason someone would want to prevent the defaults from being used?

Reply:
Some cluster operator that doesn't care about spreading and would like to skip these algorithms altogether. But it's maybe not so strong of a use case.

Reply:
defaulting this when len(...) == 0 would be more consistent with other list defaulters

Reply:
Done